tests/unit/spec/mocha.test.js

```diff
@@ -0,0 +1,319 @@
+});
+
+describe('Environment', function () {
+    it('Mocha Test should be properly configured', function () {
+        expect('test').to.equal('test');
+    });
+
+    it('should have loaded archive files', function () {
+        expect(zimArchiveFiles && zimArchiveFiles[0] && zimArchiveFiles[0].size > 0).to.be.true;
+    });
+});
+
+describe('Utils', function () {
+    it('should correctly read an IEEE_754 float from 4 bytes', function () {
+        const byteArray = new Uint8Array([194, 237, 64, 0]);
+        const float = util.readFloatFrom4Bytes(byteArray, 0);
+        expect(float).to.equal(-118.625);
+    });
+
+    it('should handle upper/lower case variations', function () {
+        const testCases = [
+            { input: 'téléphone', expected: 'Téléphone', description: 'The first letter should be uppercase' },
+            { input: 'Paris', expected: 'paris', description: 'The first letter should be lowercase' },
+            { input: 'le Couvre-chef Est sur le porte-manteaux', expected: 'Le Couvre-Chef Est Sur Le Porte-Manteaux', description: 'The first letter of every word should be uppercase' },
+            { input: 'épée', expected: 'Épée', description: 'The first letter should be uppercase (with accent)' },
+            { input: '$￥€"«xριστός» †¡Ἀνέστη!"', expected: '$￥€"«Xριστός» †¡ἀνέστη!"', description: 'First non-punctuation/non-currency Unicode letter should be uppercase, second (with breath mark) lowercase' },
+            { input: 'Καλά Νερά Μαγνησία žižek', expected: 'ΚΑΛΆ ΝΕΡΆ ΜΑΓΝΗΣΊΑ ŽIŽEK', options: 'full', description: 'All Unicode letters should be uppercase' }
+        ];
+
+        testCases.forEach(({ input, expected, options, description }) => {
+            const results = options
+                ? util.allCaseFirstLetters(input, options)
+                : util.allCaseFirstLetters(input);
+
+            expect(results, description).to.be.an('array');
+            expect(results.some(result => result === expected), description).to.be.true;
+        });
+    });
+
+    it('should remove parameters from URLs', function () {
+        const baseUrl = "A/Che cosa e l'amore?.html";
+        const testUrls = [
+            "A/Che%20cosa%20e%20l'amore%3F.html?param1=toto&param2=titi",
+            "A/Che%20cosa%20e%20l'amore%3F.html?param1=toto&param2=titi#anchor",
+            "A/Che%20cosa%20e%20l'amore%3F.html#anchor"
+        ];
+
+        testUrls.forEach(testUrl => {
+            const result = uiUtil.removeUrlParameters(testUrl);
+            // Compare normalized strings (removing special characters)
+            const normalizedResult = decodeURIComponent(result).normalize('NFD').replace(/[\u0300-\u036f]/g, '');
+            const normalizedExpected = baseUrl.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
+            expect(normalizedResult).to.equal(normalizedExpected);
+        });
+    });
+});
+
+describe('ZIM initialization', function () {
+    it('should set archive as ready', function () {
```

Comment:
For this and all other "should" below, just change to, e.g., `Set archive as ready` (etc.).